pkg/agent/endpoints/endpoints_test.go

```diff
@@ -234,15 +223,17 @@ func TestEndpoints(t *testing.T) {
 				cancel()
 				assert.NoError(t, <-errCh)
 			}()
-
+			waiForListening(t, endpoints)
```

Comment:
I wonder if this shouldn't take the errCh as well. We either expect listening or an error to happen. Then we wouldn't need the one second timeout.